### DISCUSSION TITLE
old JPA version bundled with old Hibernate ORM version

### DISCUSSION THREAD

**Discussion by arimeyer** [quarkusio/quarkus#26766] [Q&A]
I was starting to develop with Quarkus 2.10.2.Final, using Hibernate Reactive Panache, and found that many of the JPA 3.x annotations I expected to see wouldn't resolve.  I dug in and found that quarkus-hibernate-orm is bringing down JPA (jakarta.persistence-api) 2.2.3.

JPA 2.2.3 is almost 3 years old.  Why is Quarkus bundling a version this dated?  Is it possible to force the use of JPA 3.1 somehow?

Likewise, the Hibernate version is old: hibernate-core 5.6.9.Final.  Hibernate is now at 6.1.

Is there some technical reason why Quarkus is stuck on such old versions?  Is there a plan to upgrade the versions soon?

**Comment by arimeyer** [comment]
Still no response after a month.  Should I be posting this elsewhere?

**Comment by maxandersen** [comment]
Hi,

JPA 3 is part of Jakarta 10 which brings package renames across all of Java EE. Thus it is not as simple as just swapping out a single library. 

There is ongoing work to get Quarkus version that builds against Jakarta 10. You can follow that progress at https://github.com/quarkusio/quarkus/issues/21574

**Comment by arimeyer** [comment]
Thank you, Max.  It would be good for the Quarkus team to note these points in the Hibernate/Panache guides.  We need to know what our constraints are.

On that note, it would be very helpful to see, preferably very clearly on the first page of these guides, exactly which versions of standards, like JPA, JAX-RS, SmallRye, etc., that the respective Quarkus extensions depend on.  I understand that these can change rapidly, but it's something we really need to know up front.  Is there a particular place I could log this as a request?
